torch/distributed/device_mesh.py

```diff
@@ -390,6 +390,9 @@ def get_group(
             if not hasattr(self, "_dim_group_infos"):
                 raise RuntimeError("DeviceMesh process groups not initialized!")
 
+            if get_rank() not in self.mesh:
```

Comment:
@wanchaol do people use cuda tensors for device mesh? If so, we probably want to cache the values on host side to prevent a device sync here.

Reply:
That's a good point. I think most people won't pass a cuda tensor to DeviceMesh, as the mesh tensor is just a way for user to construct a mesh and it does not need cuda compute. I think we can probably audit the existing usage and ban the case when the mesh tensor is a cuda tensor. cc @wz337

Reply:
Maybe:
https://github.com/pytorch/pytorch/blob/bee24f83e7c413e142c4a99a6c37c1b1b51c7afd/torch/distributed/device_mesh.py#L201-L205
`mesh.detach().cpu()`?

Reply:
Will do this in a separate PR since it is not related to the change in this PR.